lib/src/config.cc

```diff
@@ -304,6 +310,130 @@ namespace hocon {
         return long_list;
     }
 
+    time_duration config::get_duration(string const& path) const {
+        auto v = get_value(path);
+        if (auto d = dynamic_pointer_cast<const config_double>(v)) {
+            return milliseconds(d->long_value());
+        } else if (auto l = dynamic_pointer_cast<const config_long>(v)) {
+            return milliseconds(l->long_value());
+        } else if (auto i = dynamic_pointer_cast<const config_int>(v)) {
+            return milliseconds(i->long_value());
+        } else if (auto str = dynamic_pointer_cast<const config_string>(v)) {
+            return parse_duration(str->transform_to_string(), str->origin(), path);
+        } else {
+            throw bad_value_exception(*v->origin(), path, "Value at '" + path + "' was not a number or string.");
+        }
+    }
+
+    double config::get_duration_as_double(string const& path, time_unit unit) const {
+        auto timespan = get_duration(path);
+        int64_t nanos = timespan.total_nanoseconds();
```

Comment:
There's really no conversion here? Converting nanoseconds to a 64-bit number is lossy with the config option above.